shared/src/hooks/useFetchSolVaultData.tsx

```diff
@@ -0,0 +1,78 @@
+import { useEffect, useState } from "react";
+import { useConnection } from "@solana/wallet-adapter-react";
+import { vaultUtils, vaultData } from "@zetamarkets/flex-sdk";
+import { defaultSolanaVaultData, SolanaVaultData } from "../models/vault";
+import { BigNumber } from "@ethersproject/bignumber";
+import { parseUnits } from "ethers/lib/utils";
+import { getAssetDecimals } from "../utils/asset";
+import { useFlexVault } from "./useFlexVault";
+import * as anchor from "@project-serum/anchor";
+
+const useFetchSolVaultData = (): SolanaVaultData => {
+  const { connection } = useConnection();
+  const { vault, client } = useFlexVault();
+  const [data, setData] = useState<SolanaVaultData>();
+
+  useEffect(() => {
+    const doMulticall = async () => {
+      if (!vault) return;
+
+      const [vaultAddress] = await vaultUtils.getVaultAddress("rSOL-THETA");
+      const { depositLimit, epochSequenceNumber, totalBalance, pricePerShare } =
+        await vaultData.getVaultData(connection, vaultAddress);
+
+      let totalDeposit: number = 0;
+      let totalWithdraw: number = 0;
+
+      vault.depositQueue.forEach((deposit) => {
+        const depositAmt = new anchor.BN(deposit.info.amount);
+        totalDeposit += depositAmt.toNumber();
+      });
+
+      vault.withdrawalQueue.forEach((withdraw) => {
+        const withdrawAmt = new anchor.BN(withdraw.info.amount);
+        totalWithdraw += withdrawAmt.toNumber();
+      });
```

Comment:
We can use `reduce` here to make this functional :) https://stackoverflow.com/questions/1230233/how-to-find-the-sum-of-an-array-of-numbers

Reply:
This will be amended once I finalize the structure of withdrawals